lib/db.js

```diff
@@ -36,7 +37,39 @@ const actionOneOrMany = curry(async (action, collection, isMultiple, ...params)
 });
 const insertOneOrMany = actionOneOrMany('insert');
 const updateOneOrMany = actionOneOrMany('update');
-const removeOneOrMany = actionOneOrMany('remove');
+const deleteOneOrMany = actionOneOrMany('delete');
+
+const deserialize = (data = {}) => {
+  if (Array.isArray(data)) {
+    return data.map(deserialize);
+  }
+  if (data === null) { return; }
+  if (typeof data !== 'object' || data instanceof Date) {
```

Comment:
instead of this, I'd to:

```
if(typeof data !== 'object'){
  return data;
}
if (Immutable.isImmutable(data)) {
  return data.toObject();
}
if(!isPlainObject(data)){
  return data;
}
```

And that will catch Date, as well as things like RegExp or instances of ObjectID or other weird objects we don't actually want to iterate